pysb/simulator/scipyode.py

```diff
@@ -211,9 +228,7 @@ def jacobian(t, y, p):
                     return jacmat
 
             elif self._use_inline:
-                self.jac = np.zeros(
-                    (len(self._model.odes), len(self._model.species)))
-                # Next, prepare the stringified Jacobian equations
+                # Prepare the stringified Jacobian equations.
```

Comment:
Wasn't `self.jac` technically part of the public API, although undocumented?

Reply:
No, this isn't even the symbolic Jacobian matrix. It's just a local scratch space where the `jacobian` function (analogous to the `rhs` function) stores the numeric evaluation of the Jacobian on each invocation, as an optimization over allocating a new array on each call. It has no use to external code. The same holds for `ydot` which I also cleaned up. The original implementation stored these scratch spaces as attributes of `self` and then assigned them to local variables within the integrator callback functions, but I realized they could just be locals in `__init__` and referenced directly in the callbacks. They aren't usable by external code because integrator calls to `rhs` and `jacobian` don't correspond with the currently requested time step in any way.

Reply:
My suggestion was less about the semantics/usability of the variables than the syntax - instance variables without leading underscores are suggestive of a public API. I think they're OK to drop in this case, but we should be careful about this in new code (and preferably, declare all instance variables in `__init__`; at least non-underscored ones).

Reply:
Gotcha, agreed.